### PR TITLE
Bulk add products 

### PR DESCRIPTION
**Pull request by 0x5CE**
Added the option to export products to Sakura by calling its API and supplying it with the products in JSON.

### CODE REVIEW

File: src/sakura.php [hunk 1]
```diff
@@ -731,3 +731,62 @@ public function sakura_widget_key_callback() {
 
 if ( is_admin() )
 	$sakura_network = new SakuraNetwork();
+
+
+add_filter( 'bulk_actions-edit-product', 'register_my_bulk_actions' );
```
Comment (jingtaozf) [inline src/sakura.php]: 1. Please create a new class for products management.
2. invoke add_filter in construct.
3. create an instance of this class if necessary, just it is for SakuraNetwork.

File: src/sakura.php [hunk 2]
```diff
@@ -731,3 +731,62 @@ public function sakura_widget_key_callback() {
...
+function my_bulk_action_handler( $redirect_to, $doaction, $post_ids ) {
+    
+  if ( $doaction !== 'export_to_sakura' ) {
+    return $redirect_to;
+  }
+
+  $sakura_network_options = get_option('sakura_network_option'); // Array of All Options
+  $sakura_widget_key = $sakura_network_options['sakura_widget_key']; // Sakura Widget key
+
+  $curl = curl_init('https://www.sakura.eco/api/addWCProducts');
```
Comment (jingtaozf) [inline src/sakura.php]: For a http request, please have a look of https://github.com/pegesund/sakura-wordpress#new-order
1.  $sakura_server = apply_filters('sakura_update_server_address', 'https://www.sakura.eco');
2. try wp_safe_remote_request

File: src/sakura.php [hunk 3]
```diff
@@ -731,3 +731,62 @@ public function sakura_widget_key_callback() {
...
+  curl_setopt($curl, CURLOPT_POST, 1);
+
+  $allProducts = array();
+  $payload = array();
+  $payload['token'] = 'demotoken';
```
Comment (jingtaozf) [inline src/sakura.php]: We need upgrade widget key to (widget key + widget secret) in SakuraNetwork.

File: src/sakura.php [hunk 4]
```diff
@@ -731,3 +731,62 @@ public function sakura_widget_key_callback() {
...
+  curl_close($curl);
+
+  $redirect_to = add_query_arg( 'bulk_export_posts', count( $post_ids ), $redirect_to );
+  return $redirect_to;
+}
+
+add_action( 'admin_notices', 'my_bulk_action_admin_notice' );
+ 
+function my_bulk_action_admin_notice() {
+  if ( ! empty( $_REQUEST['bulk_export_posts'] ) ) {
```
Comment (jingtaozf) [inline src/sakura.php]: Please check the response value instead of request value.

File: src/sakura.php [hunk 5]
```diff
@@ -731,3 +731,62 @@ public function sakura_widget_key_callback() {
...
+
+
+add_filter( 'bulk_actions-edit-product', 'register_my_bulk_actions' );
+ 
+function register_my_bulk_actions($bulk_actions) {
+  $bulk_actions['export_to_sakura'] = __( 'Export to Sakura', 'export_to_sakura');
+  return $bulk_actions;
+}
+
+add_filter( 'handle_bulk_actions-edit-product', 'my_bulk_action_handler', 10, 3 );
```
Comment (0x5CE) [inline src/sakura.php:743]: abc

**Review by jingtaozf — COMMENTED** [review]
Line 736:
Please create a new class for products management.
invoke add_filter in construct.
create an instance of this class if necessary, just it is for SakuraNetwork.

Line 754:
For a http request, please have a look of https://github.com/pegesund/sakura-wordpress#new-order

$sakura_server = apply_filters('sakura_update_server_address', 'https://www.sakura.eco');
try wp_safe_remote_request

Line 759:
We need upgrade widget key to (widget key + widget secret) in SakuraNetwork, and please remove demo codes here

Line 783:
Please check the response value instead of request value.

File: src/sakura.php (hunk 3 above)
Comment (0x5CE) [reply src/sakura.php]: I'll fix it in the next version, for now we need to approve it so it can be tested.

File: src/sakura.php (hunk 2 above)
Comment (0x5CE) [reply src/sakura.php]: fixed in the latest version.

File: src/sakura.php (hunk 1 above)
Comment (0x5CE) [reply src/sakura.php]: fixed in the latest version.

File: src/sakura.php (hunk 4 above)
Comment (0x5CE) [reply src/sakura.php]: fixed in the latest version.

**Pull request merged**